### PR TITLE
Add decompositions for v11 split, squeeze, and unsqueeze

### PR DESCRIPTION
**Pull request by philass**
This PR adds decompositions for V11 Split, Squeeze, and Unsqueeze to the latest versions of the ops.

Backends that don't go through KRNL will be able to leverage these decompositions. Namely  the`toch-mlir` and `mhlo` backends. (We at Groq would also be to leverage this).

This is the work of @rdicecco

Co-authored-by: Roberto DiCecco <dicecco.rob@gmail.com>
Signed-off-by: Philip Lassen <philiplassen+git@gmail.com>

### CODE REVIEW

**Comment by tungld** [comment]
@philass recommend to support ShapeOp V11 also once #1684 is landed.

**Comment by philass** [comment]
@tungld,

Right now there is no different op V11 version of Shape. Are you suggesting I introduce a new ONNX Shape op for V11 and then adding a decomposition. 

I was able to able to avoid adding a V11 op because the `end` attribute is optional, and the `start` attribute defaults to 0. Thus when a Shape from models pre opset 15 are ingested the are automatically ingested to the `ONNXShapeOp` in V15 with the correct behavior.

**Comment by tungld** [comment]
> Right now there is no different op V11 version of Shape. Are you suggesting I introduce a new ONNX Shape op for V11 and then add a decomposition. 

I see. I thought V11 was the one without start and end attributes. It is opset 13 I think: https://github.com/onnx/onnx/blob/main/docs/Changelog.md#Shape-13 that does not have the attributes.

You should change this `'Shape': [15],` in gen_onnx_mlir.py to `'Shape': [13, 15],` so that the old shape op will be imported as ONNXShapeV13Op. Otherwise, the old shape will be imported as the lastet opset.

**Comment by AlexandreEichenberger** [comment]
@tungld I thought since the newer version is only adding 2 params that defaults to the old behavior, is it still needed to have "support" for v13?

**Comment by tungld** [comment]
@philass @AlexandreEichenberger you are right. It's ok to import V13 as V15 for ShapeOp. Please ignore my previous comment and there is no need to support V13.

**Review by AlexandreEichenberger — COMMENTED** [review]
@philass @chentong319 do we still need the older versions as onnx ops? Could it be handled in the frontend? Sorry if the question is naive.

**Comment by chentong319** [comment]
> @philass @chentong319 do we still need the older versions as onnx ops? Could it be handled in the frontend? Sorry if the question is naive.

Yes, the rewriting can be done in the frontend with special import handler and then the older version op can be deleted from ONNX dialect.  But the rewriting code can not use the rule based approach. Let's keep the current approach for a while.

File: src/Transform/ONNX/Decompose.td [hunk 1]
```diff
@@ -342,6 +342,42 @@ def ClipV12Pattern : Pat<
...
+  (ONNXUnsqueezeOp $x, (CreateUnitConstant)),
+  [(AttributeIsNull:$axes)], (addBenefit 1)
+>;
+
+def UnsqueezeV11Pattern : Pat<
+  (ONNXUnsqueezeV11Op $x, $axes),
+  (ONNXUnsqueezeOp $x, (ONNXConstantOpFromDenseAttr(createDenseArrayAttr $axes))),
+  [], (addBenefit 0)
+>;
+
```
[the remark this answers is not in the document]
Comment (philass) [reply src/Transform/ONNX/Decompose.td:373]: Good point!

I've added the added

**Comment by philass** [comment]
@tungld Thanks for the review from last week.

I've added lit tests. Please let me know if you would like any other changes

**Review by tungld — APPROVED** [review]
LGTM. Thanks for addressing my comments!

**Pull request merged**